apiserver/facades/controller/caasapplicationprovisioner/provisioner.go

```diff
@@ -1253,7 +1258,14 @@ func (a *API) updateVolumeInfo(volumeUpdates map[string]volumeInfo, volumeStatus
 				return errors.Trace(err)
 			}
 		}
-
+		// Check that the machine is provisioned.
+		machineUUID, err := a.machineService.GetMachineUUID(ctx, machine.Name(volData.unitTag.Id()))
```

Comment:
This seems like a good addition but its unclear to me what its replacing? Were we not checking this before?

Reply:
Answered below for all these blocks.

Reply:
Machine? This is k8s, why are we introducing this?